src/main/scala/reporting/Converter.scala

```diff
@@ -156,37 +172,72 @@ object Converter {
-              case _                     => OtherEntry(s"$m", "not an integer literal")
+              case _                     => OtherEntry(m.toString, "not an integer literal")
             }
-          case _ => OtherEntry(s"$m", "not an integer literal")
+          case _ => OtherEntry(m.toString, "not an integer literal")
         }
       case sorts.Bool =>
         m match {
           case ConstantEntry("true")  => LitBoolEntry(true)
           case ConstantEntry("false") => LitBoolEntry(false)
-          case _ => OtherEntry(s"$m", "not a boolean literal")
+          case ConstantEntry("0") => LitBoolEntry(false) // this is kind of hacky but there is some strangeness happening...
+          case ConstantEntry("1") => LitBoolEntry(true)
+          case ApplicationEntry(_, arguments) => getConstantEntry(s, arguments.head) // sometimes bools are represented by application("=", application("var",0))
+          case _ => OtherEntry(m.toString, "not a boolean literal")
         }
-      case sorts.Seq(_) => VarEntry(m.toString, s) // will be resolved later
+      case sorts.Seq(_) 
+        | sorts.Set(_)
+        // | sorts.Map(_)
+        // | sorts.Multiset(_)
+        =>
+        // entry will be resolved on demand, i.e. when calling `extractVal` in the converter
+        // thus, we avoid dealing with recursive Refs and avoid evaluation functions at this stage
+        VarEntry(m.toString, s)
       case sorts.Perm =>
         m match {
-          case ConstantEntry(x) => LitPermEntry(x.toDouble)
+          case ConstantEntry(x) =>
+            val rational = x.toDouble match {
+              case 0.0 => Rational.zero
+              case 1.0 => Rational.one
+              case x =>
+                val intVal = x.toInt
+                val decimal = x - intVal
+                Rational(intVal, 1) + Rational(1, (1/decimal).toInt)
+            }
+            LitPermEntry(rational)
           case ApplicationEntry(name, args) =>
             val res = getConstantEntry(s, args.head)
             (res, name) match {
               case (LitPermEntry(x), "-") => LitPermEntry(-x)
-              case _                      => OtherEntry(s"$m", "not a permission literal")
+              case (_, "/") =>  (getConstantEntry(s, args.head), getConstantEntry(s, args.drop(1).head)) match {
+                case (LitPermEntry(x), LitPermEntry(y)) => LitPermEntry(x/y)
+                case _ => OtherEntry(m.toString, "not a permission literal div")
+              }
+              case _ => OtherEntry(m.toString, "not a permission literal")
             }
-          case _ => OtherEntry(s"$m", "not a permission literal")
+          case _ => OtherEntry(m.toString, "not a permission literal")
         }
+      case sorts.UserSort(id) => 
+        m match {
+          case ConstantEntry(v) => DomainValueEntry(id.toString, v.split("!").last) //this is a hack for the moment if there is any way to do this differently let me know
```

Comment:
Is the comment referring to the `.split("!")` bit? Would be better to make this explicit and searchable:
```suggestion
          // TODO: remove this string-based operation
          case ConstantEntry(v) => DomainValueEntry(id.toString, v.split("!").last)
```